app/assets/javascripts/admin/controllers/admin-badges-award.js.es6

```diff
@@ -0,0 +1,35 @@
+import Controller from "@ember/controller";
+import { ajax } from "discourse/lib/ajax";
+import { popupAjaxError } from "discourse/lib/ajax-error";
+
+export default Controller.extend({
+  saving: false,
+
+  actions: {
+    massAward() {
+      const file = document.querySelector("#massAwardCSVUpload").files[0];
+
+      if (this.model && file) {
+        const options = {
+          type: "POST",
+          processData: false,
+          contentType: false,
+          data: new FormData()
+        };
+
+        options.data.append("file", file);
+
+        this.set("saving", true);
+
+        ajax(`/admin/badges/award/${this.model.id}`, options)
+          .then(() => {
+            bootbox.alert(I18n.t("admin.badges.mass_award.success"));
```

Comment:
Maybe this should redirect to the front end view so they can see who was awarded the badge?

Reply:
The mass award will happen asynchronously, does the redirect still make sense?